From 6c462cdcfc9fcca9c060fce034c3e3507d8830b6 Mon Sep 17 00:00:00 2001
From: Kai Koehne <kai.koehne@qt.io>
Date: Mon, 30 Nov 2020 11:13:34 +0100
Subject: [PATCH 16/17] CMake: Fix handling of -extensions argument

Do not expand _lupdate_options for list commands; depending on the
content, this might cause CMake to treat it's content as separate
elements.

Also avoid IN_LIST; it was only introduced in CMake 3.3, while Qt 5.15
still also supports CMake 3.1, 3.2. Anyhow, we can as well just directly
list(FIND...).

Amends 2834bebc9e6ba9bf0577db1d7da42a52e622adc6

Change-Id: Ib4c73956477803cc8ec2f4048ffbab8daf2136e5
Reviewed-by: David Faure <david.faure@kdab.com>
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
Reviewed-by: Craig Scott <craig.scott@qt.io>
(cherry picked from commit 71ed18ed4b07c47e42f60c770bf7f5b4f8be4b20)
---
 src/linguist/Qt5LinguistToolsMacros.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/linguist/Qt5LinguistToolsMacros.cmake b/src/linguist/Qt5LinguistToolsMacros.cmake
index 3138aae4..20fb1493 100644
--- a/src/linguist/Qt5LinguistToolsMacros.cmake
+++ b/src/linguist/Qt5LinguistToolsMacros.cmake
@@ -41,10 +41,10 @@ function(QT5_CREATE_TRANSLATION _qm_files)
     set(_lupdate_files ${_LUPDATE_UNPARSED_ARGUMENTS})
     set(_lupdate_options ${_LUPDATE_OPTIONS})
 
-    if("-extensions" IN_LIST _lupdate_options)
-        list(FIND ${_lupdate_options} "-extensions" _extensions_index)
+    list(FIND _lupdate_options "-extensions" _extensions_index)
+    if(_extensions_index GREATER -1)
         math(EXPR _extensions_index "${_extensions_index} + 1")
-        list(GET ${_lupdate_options} ${_extensions_index} _extensions_list)
+        list(GET _lupdate_options ${_extensions_index} _extensions_list)
         string(REPLACE "," ";" _extensions_list "${_extensions_list}")
         list(TRANSFORM _extensions_list STRIP)
         list(TRANSFORM _extensions_list REPLACE "^\." "")
-- 
2.31.1

